patches/tModLoader/Terraria/ModLoader/GlobalTile.cs

```diff
@@ -316,4 +317,15 @@ public virtual bool CanReplace(int i, int j, int type, int tileTypeBeingPlaced)
 	public virtual void PostSetupTileMerge()
 	{
 	}
+
+	/// <summary>
+	/// This hook runs when any tree is shook. This allows you to add drops when you shake trees.
+	/// </summary>
+	/// <param name="x">The x tile coordinate of the tree.</param>
+	/// <param name="y">The y tile coordinate of the top of the tree.</param>
+	/// <param name="treeType">The type of tree that is being shaken. Modded trees will be <see cref="TreeTypes.ModTree"/> or <see cref="TreeTypes.ModPalmTree"/></param>
+	public virtual void ShakeTree(int x, int y, TreeTypes treeType)
```

Comment:
Should we supply the `ITree` instance as a parameter for modded trees? Otherwise, an example that uses the hook to target `ExampleTree` specifically is needed.